pkg/dt/node.go

```diff
@@ -120,6 +142,17 @@ func (n *Node) LookProperty(name string) (*Property, bool) {
 	return nil, false
 }
 
+// RemoveSubTreeAtIndex deletes the child at the given index (and all its children).
+func (n *Node) RemoveSubTreeAtIndex(idx int) error {
+	if idx < 0 || idx >= len(n.Children) {
+		return errInvalidChildIndex
```

Comment:
could you return:
return fmt.Errorf("idx %d is not in range 0..%d:%w", idx, len(n.Children), errInvalidChildIndex)
Then humans can see the error with more information and since you used errors.Is in the test, the test will still work.

Reply:
thank you! updated.